src/components/CommentCard.jsx

```diff
@@ -1,18 +1,35 @@
-import React, { useState } from 'react'
+import React, { useState , useContext } from 'react'
 import "../styling/commentcard.css"
 import SortDateBy from '../utils/sortDate'
+import { MdDelete } from "react-icons/md";
+import UserContext from './context/UserContext';
+import {DeleteCommentById} from "../app"
 
-function CommentCard({comment, setUpVote}) {
- 
 
-  return (
+function CommentCard({comment, setUpVote, onDeleteComment}) {
+  const {userLoggedIn} = useContext(UserContext)
+  const [err, setErr] = useState(null);
+  
+  const deleteComment = (event) => {
+    event.preventDefault();
+
+    if (comment.author === userLoggedIn.username) {
+      DeleteCommentById(comment.comment_id)
+        .catch((error) => {
+          setErr('Something went wrong, please try again.');
+        });
+        onDeleteComment(comment.comment_id);
```

Comment:
as this is not inside the .then() this will happen whether the request is successful or not - which is fine for optimistically rendering the delete but needs to be undone if the request does fail